src/main/java/org/apache/onami/persist/EntityManagerProviderImpl.java

```diff
@@ -105,6 +112,36 @@ private EntityManager createEntityManager() {
   /**
    * {@inheritDoc}
    */
+  @Override
+  public TransactionHolder withTransaction(final TransactionFacade transactionFacade) {
+    final TransactionFacade original = transactionFacades.get();
+    transactionFacades.set(transactionFacade);
+    return new TransactionHolder() {
+      @Override
+      public TransactionFacade getTransaction() {
+        return transactionFacade;
+      }
+
+      @Override
+      public void close() throws Exception {
+        transactionFacades.set(original);
```

Comment:
there is something off by this design using thread local for jpa nested transaction. Let me try to vaguely point to it. We may not need to worry, but being pedantic 

TransactionScope  {
   ChildTransactionScope {
       GrandChildTransactionScope {
           While in the scope if you add onCommit hook, it will get added to the GrandChild. What if the intent is to add to the child?
       }  
   }
}   
}